wpilib/wpilib/shuffleboard/component.py

```diff
@@ -13,8 +13,6 @@ class ShuffleboardComponent:
     def __init__(
         self, parent: ShuffleboardContainer, title: str, type: Optional[str] = None
     ):
-        assert parent is not None, "parent cannot be None"
```

Comment:
why remove this?

Reply:
> Clean up unnecessary None checks. They only exist in Java because null is easy to accidentally create.

If the type signature is followed correctly, it should be obvious that you should never pass None in. To pass None, you'd have to explicitly state so.